src/main/java/com/epam/emotionalhelp/service/impl/QuizResultServiceImpl.java

```diff
@@ -126,38 +124,39 @@ private List<SubcategoryContainerDto> initSubcategoryList(QuizResult quizResult,
         return data.stream().map(subcategoryMapper()).collect(Collectors.toCollection(ArrayList::new));
     }
 
-    private Map<EmotionCategory, List<EmotionDto>> filterByEmotions(final List<EmotionDto> inputEmotions) {
+    private Map<Emotion, List<EmotionDto>> filterByEmotions(final List<EmotionDto> inputEmotions) {
         // 1. Fill map with EMOTIONS and its SUBCATEGORIES
-        final Map<EmotionCategory, List<EmotionDto>> result = new HashMap<>();
+        final Map<Emotion, List<EmotionDto>> result = new HashMap<>();
 
         // 2. Limit input emotions by 6 maximum
-        var emotions = new ArrayList<>(inputEmotions.subList(0, LIST_SLICE_SIZE));
-        emotions.forEach(emotionDto -> {
-            // find emotion and extract description
+        var emotionsList = inputEmotions.size() > 6 ? new ArrayList<>(inputEmotions.subList(0, LIST_SLICE_SIZE)) : inputEmotions;
```

Comment:
```suggestion
        var emotionsList = new ArrayList<>(inputEmotions.subList(0, LIST_SLICE_SIZE));
```
Can we consider this code optimization?